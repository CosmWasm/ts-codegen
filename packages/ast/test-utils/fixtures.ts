// @ts-nocheck
import { JSONSchema } from '../src/types';

import canExecuteRelayResponseData from '../../../__fixtures__/vectis/govec/can_execute_relay_response.json';
import cosmosMsgForEmptyData from '../../../__fixtures__/vectis/govec/cosmos_msg_for__empty.json';
import executeMsgForEmptyData from '../../../__fixtures__/vectis/govec/execute_msg_for__empty.json';
import infoResponseData from '../../../__fixtures__/vectis/govec/info_response.json';
import relayTransactionData from '../../../__fixtures__/vectis/govec/relay_transaction.json';

import nftIdlVersionData from '../../../__fixtures__/idl-version/accounts-nft/account-nft.json';

export const nftIdlVersion: JSONSchema = nftIdlVersionData;
export const vectisCanExecuteRelayResponse: JSONSchema = canExecuteRelayResponseData;
export const vectisCosmosMsgForEmpty: JSONSchema = cosmosMsgForEmptyData;
export const vectisExecuteMsgForEmpty: JSONSchema = executeMsgForEmptyData;
export const vectisInfoResponse: JSONSchema = infoResponseData;
export const vectisRelayTransaction: JSONSchema = relayTransactionData;
