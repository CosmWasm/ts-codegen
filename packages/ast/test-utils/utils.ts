import generate from '@babel/generator';
import { RenderOptions } from '@cosmwasm/ts-codegen-ast/types';
import { ExecuteMsg, IDLObject, JSONSchema, QueryMsg } from '@cosmwasm/ts-codegen-types';
import { readFileSync } from 'fs';
import { sync as glob } from 'glob';
import { join } from 'path';

import { RenderContext } from '../src/context';

export const expectCode = (ast: any): void => {
  expect(
    generate(ast).code
  ).toMatchSnapshot();
};

export const printCode = (ast: any): void => {
  console.log(
    generate(ast).code
  );
};

export const makeContext = (
  schema: JSONSchema,
  options?: RenderOptions,
  responses?: Record<string, JSONSchema>
) => {
  return new RenderContext({
    schemas: [schema],
    responses
  }, options);
};

interface GlobContractLegacy {
  name: `/${string}.json`;
  content: JSONSchema;
}

interface GlobContract {
  name: `/${string}.json`;
  content: IDLObject;
}

const fixtureDir = join(__dirname, '../../../__fixtures__');
const globCache: Record<string, GlobContractLegacy[] | GlobContract[]> = {};

export const globIdlBasedContracts = (p: string): GlobContract[] => {
  if (globCache[p]) return globCache[p] as GlobContract[];
  // @ts-ignore
  const contracts: GlobContract[] = glob(join(fixtureDir, p, '/*.json')).map(file => {
    return {
      name: file.split(join('__fixtures__', p))[1],
      content: JSON.parse(readFileSync(file, 'utf-8'))
    };
  });
  globCache[p] = contracts;
  return contracts;
};

export const globLegacyContracts = (p: string): GlobContractLegacy[] => {
  if (globCache[p]) return globCache[p] as GlobContractLegacy[];
  // @ts-ignore
  const contracts: GlobContractLegacy[] = glob(join(fixtureDir, p, '/*.json')).map(file => {
    return {
      name: file.split(join('__fixtures__', p))[1],
      content: JSON.parse(readFileSync(file, 'utf-8'))
    };
  });
  globCache[p] = contracts;
  return contracts;
};

export const getMsgExecuteLegacyFixture = (
  p: string,
  name: `/${string}.json`
): ExecuteMsg => {
  const contracts = globLegacyContracts(p);
  const contract = contracts.find(a => a.name === name);
  return contract.content as ExecuteMsg;
};

export const getMsgQueryLegacyFixture = (
  p: string,
  name: `/${string}.json`
): QueryMsg => {
  const contracts = globLegacyContracts(p);
  const contract = contracts.find(a => a.name === name);
  return contract.content as QueryMsg;
};

export const getLegacyFixture = (
  p: string,
  name: `/${string}.json`
): JSONSchema => {
  const contracts = globLegacyContracts(p);
  const contract = contracts.find(a => a.name === name);
  return contract.content as JSONSchema;
};
