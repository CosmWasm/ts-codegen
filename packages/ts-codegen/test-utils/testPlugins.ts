import {
  BuilderContext,
  ContractInfo,
  RenderContext,
  RenderContextBase,
  RenderOptions,
} from "wasm-ast-types";
import { ClientPlugin, MessageComposerPlugin, TSBuilderOptions } from "../src";
import { ContractsContextProviderPlugin } from "../src/plugins/provider";
import { ContractsProviderBundlePlugin } from "../src/plugins/provider-bundle";

export class TestClientPlugin extends ClientPlugin {
  builderContext: BuilderContext;

  constructor(opts: RenderOptions, builderContext: BuilderContext) {
    super(opts);
    this.builderContext = builderContext;
  }

  initContext(
    contract: ContractInfo,
    options?: RenderOptions
  ): RenderContextBase<RenderOptions> {
    return new RenderContext(contract, options, this.builderContext);
  }
}

export class TestMessageComposerPlugin extends MessageComposerPlugin {
  builderContext: BuilderContext;

  constructor(opts: RenderOptions, builderContext: BuilderContext) {
    super(opts);
    this.builderContext = builderContext;
  }

  initContext(
    contract: ContractInfo,
    options?: RenderOptions
  ): RenderContextBase<RenderOptions> {
    return new RenderContext(contract, options, this.builderContext);
  }
}

export class TestContractsContextProviderPlugin extends ContractsContextProviderPlugin {
  builderContext: BuilderContext;

  constructor(opts: TSBuilderOptions, builderContext: BuilderContext) {
    super(opts);
    this.builderContext = builderContext;
  }

  initContext(
    contract: ContractInfo,
    options?: TSBuilderOptions
  ): RenderContextBase<TSBuilderOptions> {
    return new RenderContext(contract, options, this.builderContext);
  }
}

export class TestContractsProviderBundlePlugin extends ContractsProviderBundlePlugin {
  builderContext: BuilderContext;

  constructor(opts: TSBuilderOptions, builderContext: BuilderContext) {
    super(opts);
    this.builderContext = builderContext;
  }

  initContext(
    contract: ContractInfo,
    options?: TSBuilderOptions
  ): RenderContextBase<TSBuilderOptions> {
    return new RenderContext(contract, options, this.builderContext);
  }
}
