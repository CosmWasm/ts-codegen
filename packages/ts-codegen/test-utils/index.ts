import {
  BuilderContext,
  ContractInfo,
  ReactQueryOptions,
  RenderOptions,
  TSTypesOptions,
} from "wasm-ast-types";
import { ReactQueryPlugin } from "../src/plugins/react-query";
import { TypesPlugin } from "../src/plugins/types";
import { ClientPlugin } from "../src/plugins/client";
import { MessageComposerPlugin } from "../src/plugins/message-composer";
import { RecoilPlugin } from "../src/plugins/recoil";
import { MessageBuilderPlugin } from "../src/plugins/message-builder";
import { ContractsContextProviderPlugin } from "../src/plugins/provider";
import { ContractsProviderBundlePlugin } from "../src/plugins/provider-bundle";
import { TSBuilderOptions } from "../src/builder";
import { createDefaultContractInfo } from "../src/utils/contracts";
import {
  TestContractsContextProviderPlugin,
  TestClientPlugin,
  TestMessageComposerPlugin,
  TestContractsProviderBundlePlugin,
} from "./testPlugins";
import deepmerge from "deepmerge";

export const testDefaultOptions: RenderOptions = {
  enabled: true,
  types: {
    enabled: true,
    itemsUseTuples: false,
    aliasExecuteMsg: false,
  },
  client: {
    enabled: true,
    execExtendsQuery: true,
    noImplicitOverride: false,
  },
  recoil: {
    enabled: true,
  },
  messageComposer: {
    enabled: true,
  },
  messageBuilder: {
    enabled: true,
  },
  reactQuery: {
    enabled: true,
    optionalClient: false,
    version: "v4",
    mutations: false,
    camelize: true,
    queryKeys: false,
  },
  useContractsHook: {
    enabled: true,
  },
};

export async function generateReactQuery(
  contractName: string,
  contractInfo: ContractInfo,
  outPath: string,
  opts?: ReactQueryOptions
) {
  const options = opts
    ? deepmerge(testDefaultOptions, { reactQuery: opts })
    : testDefaultOptions;

  await new ReactQueryPlugin(options).render(
    outPath,
    contractName,
    contractInfo
  );
}

export async function generateTypes(
  contractName: string,
  contractInfo: ContractInfo,
  outPath: string,
  opts?: TSTypesOptions
) {
  await new TypesPlugin(opts ?? testDefaultOptions).render(
    outPath,
    contractName,
    contractInfo
  );
}

export async function generateClient(
  contractName: string,
  contractInfo: ContractInfo,
  outPath: string,
  builderContext?: BuilderContext,
  opts?: RenderOptions
) {
  if (builderContext) {
    await new TestClientPlugin(
      opts ?? testDefaultOptions,
      builderContext
    ).render(outPath, contractName, contractInfo);
  } else {
    await new ClientPlugin(opts ?? testDefaultOptions).render(
      outPath,
      contractName,
      contractInfo
    );
  }
}

export async function generateMessageComposer(
  contractName: string,
  contractInfo: ContractInfo,
  outPath: string,
  builderContext?: BuilderContext,
  opts?: RenderOptions
) {
  if (builderContext) {
    await new TestMessageComposerPlugin(
      opts ?? testDefaultOptions,
      builderContext
    ).render(outPath, contractName, contractInfo);
  } else {
    await new MessageComposerPlugin(opts ?? testDefaultOptions).render(
      outPath,
      contractName,
      contractInfo
    );
  }
}

export async function generateMessageBuilder(
  contractName: string,
  contractInfo: ContractInfo,
  outPath: string,
  opts?: RenderOptions
) {
  await new MessageBuilderPlugin(opts ?? testDefaultOptions).render(
    outPath,
    contractName,
    contractInfo
  );
}

export async function generateRecoil(
  contractName: string,
  contractInfo: ContractInfo,
  outPath: string,
  opts?: RenderOptions
) {
  await new RecoilPlugin(opts ?? testDefaultOptions).render(
    outPath,
    contractName,
    contractInfo
  );
}

export async function generateContractHooks(
  contractName: string,
  contractInfo: ContractInfo,
  outPath: string,
  builderContext?: BuilderContext
) {
  await new TestContractsContextProviderPlugin(
    {
      enabled: true,
      useShorthandCtor: true,
      useContractsHook: {
        enabled: true,
      },
    },
    builderContext
  ).render(outPath, contractName, contractInfo);

  await new TestContractsProviderBundlePlugin(
    {
      enabled: true,
      useShorthandCtor: true,
      useContractsHook: {
        enabled: true,
      },
    },
    builderContext
  ).render(outPath, undefined, contractInfo ?? createDefaultContractInfo());
}
