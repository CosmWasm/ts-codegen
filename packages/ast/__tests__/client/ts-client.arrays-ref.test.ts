import {
    createQueryClass,
    createExecuteClass,
    createExecuteInterface,
    createTypeInterface
} from '../../src'
import { expectCode, getMsgExecuteLegacyFixture, getMsgQueryLegacyFixture, makeContext } from '../../test-utils';

const queryMsg = getMsgQueryLegacyFixture('misc/schema', '/arrays-ref.json');
const execMsg = getMsgExecuteLegacyFixture('misc/schema', '/arrays-ref.json');
const ctx = makeContext(queryMsg);

it('execute_msg_for__empty', () => {
    expectCode(createTypeInterface(
        ctx,
        queryMsg
    ))
})


it('query classes', () => {
    expectCode(createQueryClass(
        ctx,
        'SG721QueryClient',
        'SG721ReadOnlyInstance',
        queryMsg
    ))
});

it('execute classes array types', () => {
    expectCode(createExecuteClass(
        ctx,
        'SG721Client',
        'SG721Instance',
        null,
        execMsg
    ))
});

it('execute interfaces no extends', () => {
    expectCode(createExecuteInterface(
        ctx,
        'SG721Instance',
        null,
        execMsg
    ))
});
