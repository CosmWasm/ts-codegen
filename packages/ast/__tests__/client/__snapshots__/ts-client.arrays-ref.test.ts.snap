// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`execute classes array types 1`] = `
"export class SG721Client implements SG721Instance {
  client: SigningCosmWasmClient;
  sender: string;
  contractAddress: string;
  constructor(client: SigningCosmWasmClient, sender: string, contractAddress: string) {
    this.client = client;
    this.sender = sender;
    this.contractAddress = contractAddress;
    this.createEdge = this.createEdge.bind(this);
    this.editEdge = this.editEdge.bind(this);
    this.removeEdge = this.removeEdge.bind(this);
    this.createGraph = this.createGraph.bind(this);
    this.createGraphSimplified = this.createGraphSimplified.bind(this);
    this.editGraphSimplified = this.editGraphSimplified.bind(this);
    this.removeGraph = this.removeGraph.bind(this);
    this.updateEdges = this.updateEdges.bind(this);
    this.findSavings = this.findSavings.bind(this);
    this.findSavingsInAGraph = this.findSavingsInAGraph.bind(this);
    this.reset = this.reset.bind(this);
    this.saveNetworkToFile = this.saveNetworkToFile.bind(this);
    this.createGraphFromFile = this.createGraphFromFile.bind(this);
    this.applySetOffFromFile = this.applySetOffFromFile.bind(this);
  }
  createEdge = async ({
    amount,
    creditor
  }: {
    amount: number;
    creditor: Addr;
  }, fee: number | StdFee | "auto" = "auto", memo?: string, funds_?: Coin[]): Promise<ExecuteResult> => {
    return await this.client.execute(this.sender, this.contractAddress, {
      create_edge: {
        amount,
        creditor
      }
    }, fee_, memo_, funds_);
  };
  editEdge = async ({
    amount,
    creditor,
    edgeId
  }: {
    amount: number;
    creditor: Addr;
    edgeId: number;
  }, fee: number | StdFee | "auto" = "auto", memo?: string, funds_?: Coin[]): Promise<ExecuteResult> => {
    return await this.client.execute(this.sender, this.contractAddress, {
      edit_edge: {
        amount,
        creditor,
        edge_id: edgeId
      }
    }, fee_, memo_, funds_);
  };
  removeEdge = async ({
    edgeId
  }: {
    edgeId: number;
  }, fee: number | StdFee | "auto" = "auto", memo?: string, funds_?: Coin[]): Promise<ExecuteResult> => {
    return await this.client.execute(this.sender, this.contractAddress, {
      remove_edge: {
        edge_id: edgeId
      }
    }, fee_, memo_, funds_);
  };
  createGraph = async ({
    graph
  }: {
    graph: Edge[];
  }, fee: number | StdFee | "auto" = "auto", memo?: string, funds_?: Coin[]): Promise<ExecuteResult> => {
    return await this.client.execute(this.sender, this.contractAddress, {
      create_graph: {
        graph
      }
    }, fee_, memo_, funds_);
  };
  createGraphSimplified = async ({
    graph,
    graphId
  }: {
    graph: Addr[][];
    graphId: number;
  }, fee: number | StdFee | "auto" = "auto", memo?: string, funds_?: Coin[]): Promise<ExecuteResult> => {
    return await this.client.execute(this.sender, this.contractAddress, {
      create_graph_simplified: {
        graph,
        graph_id: graphId
      }
    }, fee_, memo_, funds_);
  };
  editGraphSimplified = async ({
    graph,
    graphId
  }: {
    graph: Addr[][];
    graphId: number;
  }, fee: number | StdFee | "auto" = "auto", memo?: string, funds_?: Coin[]): Promise<ExecuteResult> => {
    return await this.client.execute(this.sender, this.contractAddress, {
      edit_graph_simplified: {
        graph,
        graph_id: graphId
      }
    }, fee_, memo_, funds_);
  };
  removeGraph = async ({
    graphId
  }: {
    graphId: number;
  }, fee: number | StdFee | "auto" = "auto", memo?: string, funds_?: Coin[]): Promise<ExecuteResult> => {
    return await this.client.execute(this.sender, this.contractAddress, {
      remove_graph: {
        graph_id: graphId
      }
    }, fee_, memo_, funds_);
  };
  updateEdges = async ({
    edges
  }: {
    edges: number[][];
  }, fee: number | StdFee | "auto" = "auto", memo?: string, funds_?: Coin[]): Promise<ExecuteResult> => {
    return await this.client.execute(this.sender, this.contractAddress, {
      update_edges: {
        edges
      }
    }, fee_, memo_, funds_);
  };
  findSavings = async (fee: number | StdFee | "auto" = "auto", memo?: string, funds_?: Coin[]): Promise<ExecuteResult> => {
    return await this.client.execute(this.sender, this.contractAddress, {
      find_savings: {}
    }, fee_, memo_, funds_);
  };
  findSavingsInAGraph = async ({
    graphId
  }: {
    graphId: number;
  }, fee: number | StdFee | "auto" = "auto", memo?: string, funds_?: Coin[]): Promise<ExecuteResult> => {
    return await this.client.execute(this.sender, this.contractAddress, {
      find_savings_in_a_graph: {
        graph_id: graphId
      }
    }, fee_, memo_, funds_);
  };
  reset = async (fee: number | StdFee | "auto" = "auto", memo?: string, funds_?: Coin[]): Promise<ExecuteResult> => {
    return await this.client.execute(this.sender, this.contractAddress, {
      reset: {}
    }, fee_, memo_, funds_);
  };
  saveNetworkToFile = async ({
    filepath
  }: {
    filepath: string;
  }, fee: number | StdFee | "auto" = "auto", memo?: string, funds_?: Coin[]): Promise<ExecuteResult> => {
    return await this.client.execute(this.sender, this.contractAddress, {
      save_network_to_file: {
        filepath
      }
    }, fee_, memo_, funds_);
  };
  createGraphFromFile = async ({
    filepath
  }: {
    filepath: string;
  }, fee: number | StdFee | "auto" = "auto", memo?: string, funds_?: Coin[]): Promise<ExecuteResult> => {
    return await this.client.execute(this.sender, this.contractAddress, {
      create_graph_from_file: {
        filepath
      }
    }, fee_, memo_, funds_);
  };
  applySetOffFromFile = async ({
    filepath
  }: {
    filepath: string;
  }, fee: number | StdFee | "auto" = "auto", memo?: string, funds_?: Coin[]): Promise<ExecuteResult> => {
    return await this.client.execute(this.sender, this.contractAddress, {
      apply_set_off_from_file: {
        filepath
      }
    }, fee_, memo_, funds_);
  };
}"
`;

exports[`execute interfaces no extends 1`] = `
"export interface SG721Instance {
  contractAddress: string;
  sender: string;
  createEdge: ({
    amount,
    creditor
  }: {
    amount: number;
    creditor: Addr;
  }, fee_?: number | StdFee | "auto", memo_?: string, funds_?: Coin[]) => Promise<ExecuteResult>;
  editEdge: ({
    amount,
    creditor,
    edgeId
  }: {
    amount: number;
    creditor: Addr;
    edgeId: number;
  }, fee_?: number | StdFee | "auto", memo_?: string, funds_?: Coin[]) => Promise<ExecuteResult>;
  removeEdge: ({
    edgeId
  }: {
    edgeId: number;
  }, fee_?: number | StdFee | "auto", memo_?: string, funds_?: Coin[]) => Promise<ExecuteResult>;
  createGraph: ({
    graph
  }: {
    graph: Edge[];
  }, fee_?: number | StdFee | "auto", memo_?: string, funds_?: Coin[]) => Promise<ExecuteResult>;
  createGraphSimplified: ({
    graph,
    graphId
  }: {
    graph: Addr[][];
    graphId: number;
  }, fee_?: number | StdFee | "auto", memo_?: string, funds_?: Coin[]) => Promise<ExecuteResult>;
  editGraphSimplified: ({
    graph,
    graphId
  }: {
    graph: Addr[][];
    graphId: number;
  }, fee_?: number | StdFee | "auto", memo_?: string, funds_?: Coin[]) => Promise<ExecuteResult>;
  removeGraph: ({
    graphId
  }: {
    graphId: number;
  }, fee_?: number | StdFee | "auto", memo_?: string, funds_?: Coin[]) => Promise<ExecuteResult>;
  updateEdges: ({
    edges
  }: {
    edges: number[][];
  }, fee_?: number | StdFee | "auto", memo_?: string, funds_?: Coin[]) => Promise<ExecuteResult>;
  findSavings: (fee_?: number | StdFee | "auto", memo_?: string, funds_?: Coin[]) => Promise<ExecuteResult>;
  findSavingsInAGraph: ({
    graphId
  }: {
    graphId: number;
  }, fee_?: number | StdFee | "auto", memo_?: string, funds_?: Coin[]) => Promise<ExecuteResult>;
  reset: (fee_?: number | StdFee | "auto", memo_?: string, funds_?: Coin[]) => Promise<ExecuteResult>;
  saveNetworkToFile: ({
    filepath
  }: {
    filepath: string;
  }, fee_?: number | StdFee | "auto", memo_?: string, funds_?: Coin[]) => Promise<ExecuteResult>;
  createGraphFromFile: ({
    filepath
  }: {
    filepath: string;
  }, fee_?: number | StdFee | "auto", memo_?: string, funds_?: Coin[]) => Promise<ExecuteResult>;
  applySetOffFromFile: ({
    filepath
  }: {
    filepath: string;
  }, fee_?: number | StdFee | "auto", memo_?: string, funds_?: Coin[]) => Promise<ExecuteResult>;
}"
`;

exports[`execute_msg_for__empty 1`] = `"export type ExecuteMsg = ExecuteMsg;"`;

exports[`query classes 1`] = `
"export class SG721QueryClient implements SG721ReadOnlyInstance {
  client: CosmWasmClient;
  contractAddress: string;
  constructor(client: CosmWasmClient, contractAddress: string) {
    this.client = client;
    this.contractAddress = contractAddress;
    this.createEdge = this.createEdge.bind(this);
    this.editEdge = this.editEdge.bind(this);
    this.removeEdge = this.removeEdge.bind(this);
    this.createGraph = this.createGraph.bind(this);
    this.createGraphSimplified = this.createGraphSimplified.bind(this);
    this.editGraphSimplified = this.editGraphSimplified.bind(this);
    this.removeGraph = this.removeGraph.bind(this);
    this.updateEdges = this.updateEdges.bind(this);
    this.findSavings = this.findSavings.bind(this);
    this.findSavingsInAGraph = this.findSavingsInAGraph.bind(this);
    this.reset = this.reset.bind(this);
    this.saveNetworkToFile = this.saveNetworkToFile.bind(this);
    this.createGraphFromFile = this.createGraphFromFile.bind(this);
    this.applySetOffFromFile = this.applySetOffFromFile.bind(this);
  }
  createEdge = async ({
    amount,
    creditor
  }: {
    amount: number;
    creditor: Addr;
  }): Promise<CreateEdgeResponse> => {
    return this.client.queryContractSmart(this.contractAddress, {
      create_edge: {
        amount,
        creditor
      }
    });
  };
  editEdge = async ({
    amount,
    creditor,
    edgeId
  }: {
    amount: number;
    creditor: Addr;
    edgeId: number;
  }): Promise<EditEdgeResponse> => {
    return this.client.queryContractSmart(this.contractAddress, {
      edit_edge: {
        amount,
        creditor,
        edge_id: edgeId
      }
    });
  };
  removeEdge = async ({
    edgeId
  }: {
    edgeId: number;
  }): Promise<RemoveEdgeResponse> => {
    return this.client.queryContractSmart(this.contractAddress, {
      remove_edge: {
        edge_id: edgeId
      }
    });
  };
  createGraph = async ({
    graph
  }: {
    graph: Edge[];
  }): Promise<CreateGraphResponse> => {
    return this.client.queryContractSmart(this.contractAddress, {
      create_graph: {
        graph
      }
    });
  };
  createGraphSimplified = async ({
    graph,
    graphId
  }: {
    graph: Addr[][];
    graphId: number;
  }): Promise<CreateGraphSimplifiedResponse> => {
    return this.client.queryContractSmart(this.contractAddress, {
      create_graph_simplified: {
        graph,
        graph_id: graphId
      }
    });
  };
  editGraphSimplified = async ({
    graph,
    graphId
  }: {
    graph: Addr[][];
    graphId: number;
  }): Promise<EditGraphSimplifiedResponse> => {
    return this.client.queryContractSmart(this.contractAddress, {
      edit_graph_simplified: {
        graph,
        graph_id: graphId
      }
    });
  };
  removeGraph = async ({
    graphId
  }: {
    graphId: number;
  }): Promise<RemoveGraphResponse> => {
    return this.client.queryContractSmart(this.contractAddress, {
      remove_graph: {
        graph_id: graphId
      }
    });
  };
  updateEdges = async ({
    edges
  }: {
    edges: number[][];
  }): Promise<UpdateEdgesResponse> => {
    return this.client.queryContractSmart(this.contractAddress, {
      update_edges: {
        edges
      }
    });
  };
  findSavings = async (): Promise<FindSavingsResponse> => {
    return this.client.queryContractSmart(this.contractAddress, {
      find_savings: {}
    });
  };
  findSavingsInAGraph = async ({
    graphId
  }: {
    graphId: number;
  }): Promise<FindSavingsInAGraphResponse> => {
    return this.client.queryContractSmart(this.contractAddress, {
      find_savings_in_a_graph: {
        graph_id: graphId
      }
    });
  };
  reset = async (): Promise<ResetResponse> => {
    return this.client.queryContractSmart(this.contractAddress, {
      reset: {}
    });
  };
  saveNetworkToFile = async ({
    filepath
  }: {
    filepath: string;
  }): Promise<SaveNetworkToFileResponse> => {
    return this.client.queryContractSmart(this.contractAddress, {
      save_network_to_file: {
        filepath
      }
    });
  };
  createGraphFromFile = async ({
    filepath
  }: {
    filepath: string;
  }): Promise<CreateGraphFromFileResponse> => {
    return this.client.queryContractSmart(this.contractAddress, {
      create_graph_from_file: {
        filepath
      }
    });
  };
  applySetOffFromFile = async ({
    filepath
  }: {
    filepath: string;
  }): Promise<ApplySetOffFromFileResponse> => {
    return this.client.queryContractSmart(this.contractAddress, {
      apply_set_off_from_file: {
        filepath
      }
    });
  };
}"
`;
